src/App.js

```diff
@@ -1,13 +1,37 @@
-import "./App.css";
-import { Footer, Navbar } from "./frontend/components";
-import { LandingPage } from "./frontend/pages";
+import { useLocation } from "react-router-dom";
+
+import { Footer, Navbar, Sidebar } from "./frontend/components";
 import { Router } from "./frontend/routing/Router";
 
+import "./App.css";
+
+const sidebarRestrict = (pathname) => {
+  if ((pathname === "/") || (pathname === "/signin") || (pathname === "/signup")) {
+    return false;
+  } else {
+    return true;
+  }
+}
+
+const flexToggle = (pathname) => {
+  if (sidebarRestrict(pathname)) {
+    return "flex--row";
+  }
+  else {
+    return "";
+  }
+}
+
 function App() {
+  const location = useLocation();
+
   return (
     <div className="App">
-      <Navbar />
-      <Router />
+      <Navbar searchRestrict={sidebarRestrict} />
+      <div className={flexToggle(location.pathname)}>
+        {sidebarRestrict(location.pathname) && <Sidebar />}
+        <Router />
+      </div>
       <Footer />
     </div>
   );
```

Comment:
This functionality is not very evident as to what it does. You could either give a better name or drop some comments.

Reply:
Its for the visiblity of the sidebar . i wanted the sidebar component to be on the main pages only (ie. the notes , labels , archieve and trash ) didnt want it to show on the (signin , signup and landing pages) so made a function sidebarRestrict such that it hides the sidebar by checking the location.pathname ie. which page it is on ..... similarly for the search input too